optuna/storages/_rdb/alembic/versions/v2.4.0.a.py

```diff
@@ -0,0 +1,124 @@
+            sa.ForeignKeyConstraint(
+                ["study_id"],
+                ["studies.study_id"],
+            ),
+            sa.PrimaryKeyConstraint("study_direction_id"),
+            sa.UniqueConstraint("study_id", "objective"),
+        )
+
+    if "trial_intermediate_values" not in tables:
+        op.create_table(
+            "trial_intermediate_values",
+            sa.Column("trial_intermediate_values_id", sa.Integer(), nullable=False),
+            sa.Column("trial_id", sa.Integer(), nullable=True),
+            sa.Column("step", sa.Integer(), nullable=True),
+            sa.Column("value", sa.Float(), nullable=True),
+            sa.ForeignKeyConstraint(
+                ["trial_id"],
+                ["trials.trial_id"],
+            ),
+            sa.PrimaryKeyConstraint("trial_intermediate_values_id"),
+            sa.UniqueConstraint("trial_id", "step"),
+        )
+
+    with op.batch_alter_table("trial_values", schema=None) as batch_op:
+        batch_op.add_column(sa.Column("objective", sa.Integer(), nullable=True))
+        batch_op.create_unique_constraint("value_constraint", ["trial_id", "objective"])
+
+    session = orm.Session(bind=bind)
+    try:
+
+        class BackwardCompatibleStudyModel(StudyModel):
+            direction = sa.Column(sa.Enum(StudyDirection))
+
+        class BackwardCompatibleTrialValueModel(TrialValueModel):
+            step = sa.Column(sa.Integer)
+
+        class BackwardCompatibleTrialModel(TrialModel):
+            value = sa.Column(sa.Float)
+
+        studies_records = session.query(BackwardCompatibleStudyModel).all()
+        objects = [
+            StudyDirectionModel(study_id=r.study_id, direction=r.direction, objective=0)
+            for r in studies_records
+        ]
+        session.bulk_save_objects(objects)
+
+        intermediate_values_records = session.query(BackwardCompatibleTrialValueModel).all()
+        objects = [
+            TrialIntermediateValueModel(trial_id=r.trial_id, value=r.value, step=r.step)
+            for r in intermediate_values_records
+        ]
+        session.bulk_save_objects(objects)
+
+        session.query(TrialValueModel).delete()
+        trials_records = session.query(BackwardCompatibleTrialModel).all()
+        objects = [
+            TrialValueModel(trial_id=r.trial_id, value=r.value, objective=0)
+            for r in trials_records
+        ]
+        session.bulk_save_objects(objects)
+
+        session.commit()
+    except SQLAlchemyError as e:
+        session.rollback()
+        raise e
+    finally:
+        session.close()
+
+    with op.batch_alter_table("studies", schema=None) as batch_op:
+        batch_op.drop_column("direction")
+
+    with op.batch_alter_table("trial_values", schema=None) as batch_op:
+        batch_op.drop_column("step")
+
+    with op.batch_alter_table("trials", schema=None) as batch_op:
+        batch_op.drop_column("value")
+
+
+def downgrade():
+    pass
```

Comment:
I removed the downgrade implementation in this PR since we decide the downgrade method is out of the scope of this PR from the internal discussion. I will work on this in another PR.

Reply:
Sounds reasonable given that we'd have to write migration code that mimics the currently trick to allow MO on top of the SO storage interfaces/models.

Reply:
A detail but how about leaving a TODO or explain why it's skipped in the code?